### PR TITLE
Add SFT/PEFT HF tests

### PR DESCRIPTION
**Pull request by akoumpa**
# What does this PR do ?

Add CI tests for PEFT/SFT with HF models for single gpu and DDP setups.

TODO (pending on hf ckpt work for perf): tests for saving / resuming from ckpt. 

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: .github/workflows/cicd-main.yml [hunk 1]
```diff
@@ -3574,17 +3574,60 @@ jobs:
...
 
 
+  L2_HF_Transformer_PEFT:
+    needs: [ cicd-test-container-setup ]
+    uses: ./.github/workflows/_test_template.yml
+    if: contains(fromJSON(needs.cicd-test-container-setup.outputs.test_to_run), 'L2_HF_Transformer_PEFT') || needs.cicd-test-container-setup.outputs.all == 'true'
+    with:
+      RUNNER: self-hosted-azure
+      SCRIPT: |
+        python tests/collections/llm/hf/peft.py --model /home/TestData/nlp/hf_gemma/hf_gemma_2b --max-steps 10
```
Comment (HuiyingLi) [inline .github/workflows/cicd-main.yml:3584]: I needed to add `--strategy ddp` to use  `torch.distributed.get_rank()` while saving ckpt.

File: examples/llm/peft/hf.py [hunk 2]
```diff
@@ -39,14 +39,17 @@ def formatting_prompts_func(examples):
...
-            'tokens': tokens,
-            'labels': tokens[1:] + [tokens[-1]],
-        }
+        ans['labels'] = ans['input_ids']
+        return ans
 
-    datamodule = llm.HFDatasetDataModule("rajpurkar/squad", split="train", pad_token_id=tokenizer.eos_token_id)
-    datamodule.map(formatting_prompts_func, batched=False, batch_size=2)
+    tokenizer = getattr(tokenizer, 'tokenizer', tokenizer)
+    datamodule = llm.HFDatasetDataModule("rajpurkar/squad", split="train[:100]", pad_token_id=tokenizer.eos_token_id)
```
Comment (HuiyingLi) [inline examples/llm/peft/hf.py:46]: "train[:100]` doesn't match the "train" split name later on

File: .github/workflows/cicd-main.yml (hunk 1 above)
Comment (akoumpa) [reply .github/workflows/cicd-main.yml:3584]: @HuiyingLi thank you, can you push your fix to this branch as well?

File: examples/llm/peft/hf.py (hunk 2 above)
Comment (akoumpa) [reply examples/llm/peft/hf.py:46]: I'm working on this now.

File: tests/collections/llm/hf/peft.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        output = examples["answers"]['text']
+        if isinstance(output, list):
+            output = output[0]
+        text = alpaca_prompt.format(instruction, input, output) + EOS_TOKEN
+        ans = tokenizer(text)
+        ans['labels'] = ans['input_ids']
+        return ans
+
+    tokenizer = getattr(tokenizer, 'tokenizer', tokenizer)
+    datamodule = llm.HFDatasetDataModule(data_path, split="train[:100]", pad_token_id=tokenizer.eos_token_id)
```
Comment (HuiyingLi) [inline tests/collections/llm/hf/peft.py:48]: same here

File: .github/workflows/cicd-main.yml (hunk 1 above)
Comment (akoumpa) [reply .github/workflows/cicd-main.yml:3584]: Added --disable-ckpt option for the time being and we can remove it once the checkpoint work is done.

File: .github/workflows/cicd-main.yml [hunk 4]
```diff
@@ -3574,17 +3574,60 @@ jobs:
         inference.outfile_path=/tmp/nlp_mcore_t5_lora_tuning_tp2/out.jsonl
 
 
+  L2_HF_Transformer_PEFT:
+    needs: [ cicd-test-container-setup ]
+    uses: ./.github/workflows/_test_template.yml
+    if: contains(fromJSON(needs.cicd-test-container-setup.outputs.test_to_run), 'L2_HF_Transformer_PEFT') || needs.cicd-test-container-setup.outputs.all == 'true'
+    with:
+      RUNNER: self-hosted-azure
```
[the remark this answers is not in the document]
Comment (akoumpa) [reply .github/workflows/cicd-main.yml:3582]: that works too.

**Review by HuiyingLi — APPROVED** [review]
LGTM!

**Review by ko3n1g — APPROVED** [review]
Also the selective run after changing 1-gpu runners passed: https://github.com/NVIDIA/NeMo/actions/runs/12301650348/job/34334351238

**Pull request merged**